### ISSUE TITLE
Nx Migrate - No such file or directory error

### ISSUE THREAD

**Issue by andy-zhou** [nrwl/nx#5385]
## Current Behavior
I'm trying to upgrade with `yarn nx migrate latest`, but I'm encountering the following error:

```
/bin/sh: /var/folders/dd/w_m3kf3x1czdw8rh8r6j72lc0000gn/T/tmp-49806-3YOh2gWVBr05/node_modules/.bin/tao: No such file or directory
<directory>/node_modules/yargs/build/lib/yargs.js:1132
                throw err;
                ^

Error: Command failed: /var/folders/dd/w_m3kf3x1czdw8rh8r6j72lc0000gn/T/tmp-49806-3YOh2gWVBr05/node_modules/.bin/tao migrate --run-migrations
    at checkExecSyncError (child_process.js:616:11)
    at Object.execSync (child_process.js:652:15)
    at Object.handler (<directory>/node_modules/@nrwl/workspace/src/command-line/nx-commands.js:87:25)
    at Object.runCommand (<directory>/node_modules/yargs/build/lib/command.js:196:48)
    at Object.parseArgs [as _parseArgs] (<directory>/node_modules/yargs/build/lib/yargs.js:1043:55)
    at Object.get [as argv] (<directory>/node_modules/yargs/build/lib/yargs.js:986:25)
    at Object.initLocal (<directory>/node_modules/@nrwl/cli/lib/init-local.js:24:13)
    at Object.<anonymous> (<directory>/node_modules/@nrwl/cli/bin/nx.js:43:18)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10) {
  status: 127,
  signal: null,
  output: [ null, null, null ],
  pid: 49816,
  stdout: null,
  stderr: null
}
```

I've also tried the following which gives me the same error: 

```sh
$ npx @nrwl/tao@latest migrate latest
<...>
The migrate command has run successfully.
<...>

$ yarn
<Installation output>

$ nx migrate --run-migrations
<Error Here>

```

## Expected Behavior
A successful migration

## More Info
I tried creating a new nx repo with `npx create-nx-workspace@11.1.5`. Running `yarn nx migrate latest` works here. A reasonable solution would be to to just migrate my code to the new repository.

### Environment
Before migrating with `@nrwl/tao@latest`
```sh
>  NX  Report complete - copy this into the issue template

  Node : 14.16.0
  OS   : darwin x64
  yarn : 2.4.0
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 11.1.5
  @nrwl/cypress : 11.1.5
  @nrwl/devkit : 11.1.5
  @nrwl/eslint-plugin-nx : 11.1.5
  @nrwl/express : 11.1.5
  @nrwl/jest : 11.1.5
  @nrwl/linter : 11.1.5
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : 11.1.5
  @nrwl/react : 11.1.5
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.1.5
  @nrwl/web : 11.1.5
  @nrwl/workspace : 11.1.5
  typescript : 4.1.5
```

After migrating with `@nrwl/tao@latest`
```sh
>  NX  Report complete - copy this into the issue template
  Node : 14.16.0
  OS   : darwin x64
  yarn : 2.4.0
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 12.0.7
  @nrwl/cypress : 12.0.7
  @nrwl/devkit : 12.0.7
  @nrwl/eslint-plugin-nx : 12.0.7
  @nrwl/express : 12.0.7
  @nrwl/jest : 12.0.7
  @nrwl/linter : 12.0.7
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : 12.0.7
  @nrwl/react : 12.0.7
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.0.7
  @nrwl/web : 12.0.7
  @nrwl/workspace : 12.0.7
  @nrwl/storybook : 12.0.7
  @nrwl/gatsby : Not Found
  typescript : 4.1.4
```

I'm using an M1 Mac, yarn 2 (node_modules linker, not pnp), and the [`asdf` version manager](https://github.com/asdf-vm/asdf).

**Comment by andy-zhou** [comment]
I did some more digging - this appears to be an issue with `yarn 2` defaulting to pnp when installing in the temporary directory for `tao`. I solved this by adding a global `~/.yarnrc.yml` with the contents `nodeLinker: node-modules`.

It may be worth it to just add this `.yarnrc.yml` file to the temporary directory in case users are using yarn 2.

I'm still not sure why it works for a new directory.
